### ISSUE TITLE
repeat option fails in workloads with input or output option 

### ISSUE THREAD

**Issue by Brandonage** [SparkTC/spark-bench#147]
### Spark-Bench Version
spark-bench_2.1.1_0.2.2-RELEASE_55.tgz

### Spark Version on Your Cluster
Spark 2.1.0

### Scala Version on Your Spark Cluster
2.11. Not sure but I think is not relevant

### Your Exact Configuration File (with system details anonymized)

```spark-bench = {
  spark-submit-config = [{
    spark-args = {
        executor-memory = "2g"
    }
    suites-parallel = false
    workload-suites = [
      {
        descr = "Generate a dataset, then take that same dataset and write it out to Parquet format"
        benchmark-output = "hdfs:///tmp/csv-vs-parquet/results-data-gen.csv"
        // We need to generate the dataset first through the data generator, then we take that dataset and convert it to Parquet.
        parallel = false
        repeat = 10
        workloads = [
          {
            name = "data-generation-kmeans"
            rows = 10000000
            cols = 24
            output = "hdfs:///tmp/csv-vs-parquet/kmeans-data.csv"
          },
          {
            name = "sql"
            query = "select * from input"
            input = "hdfs:///tmp/csv-vs-parquet/kmeans-data.csv"
            output = "hdfs:///tmp/csv-vs-parquet/kmeans-data.parquet"
          }
        ]
      }
    ]
  }]
}```
<!-- Please surround your configuration file with three backticks ``` before and after your configuration.
This forces code formatting for markdown and makes it MUCH easier to read! -->

### Relevant Stack Trace (If Applicable)
```s on 9.0.2.131 (executor 0) (8/36)
18/01/31 16:40:59 INFO scheduler.TaskSetManager: Starting task 20.0 in stage 8.0 (TID 99, 9.0.1.131, executor 2, partition 20, ANY, 6788 bytes)
18/01/31 16:40:59 INFO scheduler.TaskSetManager: Finished task 0.0 in stage 8.0 (TID 79) in 31779 ms on 9.0.1.131 (executor 2) (9/36)
18/01/31 16:41:00 INFO scheduler.TaskSetManager: Starting task 21.0 in stage 8.0 (TID 100, 9.0.2.131, executor 0, partition 21, ANY, 6788 bytes)
18/01/31 16:41:00 INFO scheduler.TaskSetManager: Finished task 11.0 in stage 8.0 (TID 90) in 32291 ms on 9.0.2.131 (executor 0) (10/36)
18/01/31 16:41:00 INFO scheduler.TaskSetManager: Starting task 22.0 in stage 8.0 (TID 101, 9.0.2.131, executor 0, partition 22, ANY, 6788 bytes)
18/01/31 16:41:00 INFO scheduler.TaskSetManager: Finished task 5.0 in stage 8.0 (TID 84) in 32935 ms on 9.0.2.131 (executor 0) (11/36)
18/01/31 16:41:01 INFO scheduler.TaskSetManager: Starting task 23.0 in stage 8.0 (TID 102, 9.0.2.131, executor 0, partition 23, ANY, 6788 bytes)
18/01/31 16:41:01 INFO scheduler.TaskSetManager: Finished task 2.0 in stage 8.0 (TID 81) in 33251 ms on 9.0.2.131 (executor 0) (12/36)
18/01/31 16:41:20 INFO scheduler.TaskSetManager: Starting task 24.0 in stage 8.0 (TID 103, 9.0.4.132, executor 1, partition 24, ANY, 6788 bytes)
18/01/31 16:41:20 INFO scheduler.TaskSetManager: Finished task 12.0 in stage 8.0 (TID 91) in 23335 ms on 9.0.4.132 (executor 1) (13/36)
18/01/31 16:41:21 INFO scheduler.TaskSetManager: Starting task 25.0 in stage 8.0 (TID 104, 9.0.4.132, executor 1, partition 25, ANY, 6788 bytes)
18/01/31 16:41:21 INFO scheduler.TaskSetManager: Finished task 15.0 in stage 8.0 (TID 94) in 23221 ms on 9.0.4.132 (executor 1) (14/36)
18/01/31 16:41:22 INFO scheduler.TaskSetManager: Starting task 26.0 in stage 8.0 (TID 105, 9.0.4.132, executor 1, partition 26, ANY, 6788 bytes)
18/01/31 16:41:22 INFO scheduler.TaskSetManager: Finished task 13.0 in stage 8.0 (TID 92) in 24251 ms on 9.0.4.132 (executor 1) (15/36)
18/01/31 16:41:23 INFO scheduler.TaskSetManager: Starting task 27.0 in stage 8.0 (TID 106, 9.0.4.132, executor 1, partition 27, ANY, 6788 bytes)
18/01/31 16:41:23 INFO scheduler.TaskSetManager: Finished task 14.0 in stage 8.0 (TID 93) in 25277 ms on 9.0.4.132 (executor 1) (16/36)
18/01/31 16:41:24 INFO scheduler.TaskSetManager: Starting task 28.0 in stage 8.0 (TID 107, 9.0.2.131, executor 0, partition 28, ANY, 6788 bytes)
18/01/31 16:41:24 INFO scheduler.TaskSetManager: Finished task 21.0 in stage 8.0 (TID 100) in 23844 ms on 9.0.2.131 (executor 0) (17/36)
18/01/31 16:41:24 INFO scheduler.TaskSetManager: Starting task 29.0 in stage 8.0 (TID 108, 9.0.1.131, executor 2, partition 29, ANY, 6788 bytes)
18/01/31 16:41:24 INFO scheduler.TaskSetManager: Finished task 20.0 in stage 8.0 (TID 99) in 25347 ms on 9.0.1.131 (executor 2) (18/36)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Starting task 30.0 in stage 8.0 (TID 109, 9.0.2.131, executor 0, partition 30, ANY, 6788 bytes)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Finished task 22.0 in stage 8.0 (TID 101) in 26205 ms on 9.0.2.131 (executor 0) (19/36)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Starting task 31.0 in stage 8.0 (TID 110, 9.0.1.131, executor 2, partition 31, ANY, 6788 bytes)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Finished task 16.0 in stage 8.0 (TID 95) in 28330 ms on 9.0.1.131 (executor 2) (20/36)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Starting task 32.0 in stage 8.0 (TID 111, 9.0.2.131, executor 0, partition 32, ANY, 6788 bytes)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Finished task 19.0 in stage 8.0 (TID 98) in 27602 ms on 9.0.2.131 (executor 0) (21/36)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Starting task 33.0 in stage 8.0 (TID 112, 9.0.1.131, executor 2, partition 33, ANY, 6788 bytes)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Finished task 17.0 in stage 8.0 (TID 96) in 28663 ms on 9.0.1.131 (executor 2) (22/36)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Starting task 34.0 in stage 8.0 (TID 113, 9.0.1.131, executor 2, partition 34, ANY, 6788 bytes)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Finished task 18.0 in stage 8.0 (TID 97) in 28275 ms on 9.0.1.131 (executor 2) (23/36)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Starting task 35.0 in stage 8.0 (TID 114, 9.0.2.131, executor 0, partition 35, ANY, 6788 bytes)
18/01/31 16:41:27 INFO scheduler.TaskSetManager: Finished task 23.0 in stage 8.0 (TID 102) in 26392 ms on 9.0.2.131 (executor 0) (24/36)
18/01/31 16:41:41 INFO scheduler.TaskSetManager: Finished task 24.0 in stage 8.0 (TID 103) in 20059 ms on 9.0.4.132 (executor 1) (25/36)
18/01/31 16:41:43 INFO scheduler.TaskSetManager: Finished task 35.0 in stage 8.0 (TID 114) in 16392 ms on 9.0.2.131 (executor 0) (26/36)
18/01/31 16:41:44 INFO scheduler.TaskSetManager: Finished task 25.0 in stage 8.0 (TID 104) in 23136 ms on 9.0.4.132 (executor 1) (27/36)
18/01/31 16:41:45 INFO scheduler.TaskSetManager: Finished task 26.0 in stage 8.0 (TID 105) in 23259 ms on 9.0.4.132 (executor 1) (28/36)
18/01/31 16:41:45 INFO scheduler.TaskSetManager: Finished task 28.0 in stage 8.0 (TID 107) in 21494 ms on 9.0.2.131 (executor 0) (29/36)
18/01/31 16:41:45 INFO scheduler.TaskSetManager: Finished task 34.0 in stage 8.0 (TID 113) in 18255 ms on 9.0.1.131 (executor 2) (30/36)
18/01/31 16:41:46 INFO scheduler.TaskSetManager: Finished task 27.0 in stage 8.0 (TID 106) in 23175 ms on 9.0.4.132 (executor 1) (31/36)
18/01/31 16:41:47 INFO scheduler.TaskSetManager: Finished task 29.0 in stage 8.0 (TID 108) in 22654 ms on 9.0.1.131 (executor 2) (32/36)
18/01/31 16:41:48 INFO scheduler.TaskSetManager: Finished task 32.0 in stage 8.0 (TID 111) in 21226 ms on 9.0.2.131 (executor 0) (33/36)
18/01/31 16:41:48 INFO scheduler.TaskSetManager: Finished task 30.0 in stage 8.0 (TID 109) in 21514 ms on 9.0.2.131 (executor 0) (34/36)
18/01/31 16:41:49 INFO scheduler.TaskSetManager: Finished task 33.0 in stage 8.0 (TID 112) in 22264 ms on 9.0.1.131 (executor 2) (35/36)
18/01/31 16:41:50 INFO scheduler.TaskSetManager: Finished task 31.0 in stage 8.0 (TID 110) in 23186 ms on 9.0.1.131 (executor 2) (36/36)
18/01/31 16:41:50 INFO scheduler.TaskSchedulerImpl: Removed TaskSet 8.0, whose tasks have all completed, from pool
18/01/31 16:41:50 INFO scheduler.DAGScheduler: ResultStage 8 (parquet at SparkFuncs.scala:36) finished in 82.370 s
18/01/31 16:41:50 INFO scheduler.DAGScheduler: Job 8 finished: parquet at SparkFuncs.scala:36, took 82.434367 s
18/01/31 16:41:50 INFO datasources.FileFormatWriter: Job null committed.
18/01/31 16:41:50 INFO spark.SparkContext: Starting job: first at KMeansDataGen.scala:67
18/01/31 16:41:50 INFO scheduler.DAGScheduler: Got job 9 (first at KMeansDataGen.scala:67) with 1 output partitions
18/01/31 16:41:50 INFO scheduler.DAGScheduler: Final stage: ResultStage 9 (first at KMeansDataGen.scala:67)
18/01/31 16:41:50 INFO scheduler.DAGScheduler: Parents of final stage: List()
18/01/31 16:41:50 INFO scheduler.DAGScheduler: Missing parents: List()
18/01/31 16:41:50 INFO scheduler.DAGScheduler: Submitting ResultStage 9 (MapPartitionsRDD[27] at map at KMeansDataGenerator.scala:60), which has no missing parents
18/01/31 16:41:50 INFO memory.MemoryStore: Block broadcast_14 stored as values in memory (estimated size 2.4 KB, free 365.9 MB)
18/01/31 16:41:50 INFO memory.MemoryStore: Block broadcast_14_piece0 stored as bytes in memory (estimated size 1765.0 B, free 365.9 MB)
18/01/31 16:41:50 INFO storage.BlockManagerInfo: Added broadcast_14_piece0 in memory on 9.0.0.131:40591 (size: 1765.0 B, free: 366.2 MB)
18/01/31 16:41:50 INFO spark.SparkContext: Created broadcast 14 from broadcast at DAGScheduler.scala:996
18/01/31 16:41:50 INFO scheduler.DAGScheduler: Submitting 1 missing tasks from ResultStage 9 (MapPartitionsRDD[27] at map at KMeansDataGenerator.scala:60)
18/01/31 16:41:50 INFO scheduler.TaskSchedulerImpl: Adding task set 9.0 with 1 tasks
18/01/31 16:41:50 INFO scheduler.TaskSetManager: Starting task 0.0 in stage 9.0 (TID 115, 9.0.2.131, executor 0, partition 0, PROCESS_LOCAL, 6031 bytes)
18/01/31 16:41:50 INFO storage.BlockManagerInfo: Added broadcast_14_piece0 in memory on 9.0.2.131:33302 (size: 1765.0 B, free: 912.2 MB)
18/01/31 16:41:50 INFO scheduler.TaskSetManager: Finished task 0.0 in stage 9.0 (TID 115) in 20 ms on 9.0.2.131 (executor 0) (1/1)
18/01/31 16:41:50 INFO scheduler.TaskSchedulerImpl: Removed TaskSet 9.0, whose tasks have all completed, from pool
18/01/31 16:41:50 INFO scheduler.DAGScheduler: ResultStage 9 (first at KMeansDataGen.scala:67) finished in 0.020 s
18/01/31 16:41:50 INFO scheduler.DAGScheduler: Job 9 finished: first at KMeansDataGen.scala:67, took 0.025692 s
Exception in thread "main" com.ibm.sparktc.sparkbench.utils.SparkBenchException: Error: hdfs:///tmp/csv-vs-parquet/kmeans-data3.csv already exists!
	at com.ibm.sparktc.sparkbench.utils.SparkFuncs$.verifyPathNotExistsOrThrow(SparkFuncs.scala:70)
	at com.ibm.sparktc.sparkbench.utils.SparkFuncs$.writeToDisk(SparkFuncs.scala:27)
	at com.ibm.sparktc.sparkbench.datageneration.mlgenerator.KMeansDataGen$$anonfun$1.apply$mcV$sp(KMeansDataGen.scala:74)
	at com.ibm.sparktc.sparkbench.datageneration.mlgenerator.KMeansDataGen$$anonfun$1.apply(KMeansDataGen.scala:74)
	at com.ibm.sparktc.sparkbench.datageneration.mlgenerator.KMeansDataGen$$anonfun$1.apply(KMeansDataGen.scala:74)
	at com.ibm.sparktc.sparkbench.utils.GeneralFunctions$.time(GeneralFunctions.scala:67)
	at com.ibm.sparktc.sparkbench.datageneration.mlgenerator.KMeansDataGen.doWorkload(KMeansDataGen.scala:74)
	at com.ibm.sparktc.sparkbench.workload.Workload$class.run(Workload.scala:55)
	at com.ibm.sparktc.sparkbench.datageneration.mlgenerator.KMeansDataGen.run(KMeansDataGen.scala:42)
	at com.ibm.sparktc.sparkbench.workload.SuiteKickoff$$anonfun$com$ibm$sparktc$sparkbench$workload$SuiteKickoff$$runSerially$1.apply(SuiteKickoff.scala:95)
	at com.ibm.sparktc.sparkbench.workload.SuiteKickoff$$anonfun$com$ibm$sparktc$sparkbench$workload$SuiteKickoff$$runSerially$1.apply(SuiteKickoff.scala:95)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.immutable.List.map(List.scala:285)
	at com.ibm.sparktc.sparkbench.workload.SuiteKickoff$.com$ibm$sparktc$sparkbench$workload$SuiteKickoff$$runSerially(SuiteKickoff.scala:95)
	at com.ibm.sparktc.sparkbench.workload.SuiteKickoff$$anonfun$2.apply(SuiteKickoff.scala:69)
	at com.ibm.sparktc.sparkbench.workload.SuiteKickoff$$anonfun$2.apply(SuiteKickoff.scala:64)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.immutable.Range.foreach(Range.scala:160)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at com.ibm.sparktc.sparkbench.workload.SuiteKickoff$.run(SuiteKickoff.scala:64)
	at com.ibm.sparktc.sparkbench.workload.MultipleSuiteKickoff$$anonfun$com$ibm$sparktc$sparkbench$workload$MultipleSuiteKickoff$$runSuitesSerially$1.apply(MultipleSuiteKickoff.scala:41)
	at com.ibm.sparktc.sparkbench.workload.MultipleSuiteKickoff$$anonfun$com$ibm$sparktc$sparkbench$workload$MultipleSuiteKickoff$$runSuitesSerially$1.apply(MultipleSuiteKickoff.scala:41)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at com.ibm.sparktc.sparkbench.workload.MultipleSuiteKickoff$.com$ibm$sparktc$sparkbench$workload$MultipleSuiteKickoff$$runSuitesSerially(MultipleSuiteKickoff.scala:41)
	at com.ibm.sparktc.sparkbench.workload.MultipleSuiteKickoff$$anonfun$run$1.apply(MultipleSuiteKickoff.scala:30)
	at com.ibm.sparktc.sparkbench.workload.MultipleSuiteKickoff$$anonfun$run$1.apply(MultipleSuiteKickoff.scala:27)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at com.ibm.sparktc.sparkbench.workload.MultipleSuiteKickoff$.run(MultipleSuiteKickoff.scala:27)
	at com.ibm.sparktc.sparkbench.cli.CLIKickoff$.main(CLIKickoff.scala:33)
	at com.ibm.sparktc.sparkbench.cli.CLIKickoff.main(CLIKickoff.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:743)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:187)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:212)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:126)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.lang.Exception
	at com.ibm.sparktc.sparkbench.utils.SparkBenchException$.apply$default$2(SparkBenchException.scala:22)
	... 44 more
18/01/31 16:41:50 INFO spark.SparkContext: Invoking stop() from shutdown hook
18/01/31 16:41:50 INFO server.ServerConnector: Stopped Spark@173b9122{HTTP/1.1}{0.0.0.0:4040}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@22c01ab0{/stages/stage/kill,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@585811a4{/jobs/job/kill,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@3277e499{/api,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@2a492f2a{/,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@76318a7d{/static,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@56b78e55{/executors/threadDump/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@2f66e802{/executors/threadDump,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@1b58ff9e{/executors/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@641856{/executors,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@63f34b70{/environment/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@5ebd56e9{/environment,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@2427e004{/storage/rdd/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@1a6f2363{/storage/rdd,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@681aad3b{/storage/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@52350abb{/storage,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@cc6460c{/stages/pool/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@7d61eccf{/stages/pool,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@2cd2c8fe{/stages/stage/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@226f885f{/stages/stage,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@21ca139c{/stages/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@1df98368{/stages,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@77cf3f8b{/jobs/job/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@4628b1d3{/jobs/job,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@71a9b4c7{/jobs/json,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO handler.ContextHandler: Stopped o.s.j.s.ServletContextHandler@34abdee4{/jobs,null,UNAVAILABLE,@Spark}
18/01/31 16:41:50 INFO ui.SparkUI: Stopped Spark web UI at http://9.0.0.131:4040
18/01/31 16:41:50 INFO cluster.StandaloneSchedulerBackend: Shutting down all executors
18/01/31 16:41:50 INFO cluster.CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
18/01/31 16:41:50 INFO spark.MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
18/01/31 16:41:50 INFO memory.MemoryStore: MemoryStore cleared
18/01/31 16:41:50 INFO storage.BlockManager: BlockManager stopped
18/01/31 16:41:50 INFO storage.BlockManagerMaster: BlockManagerMaster stopped
18/01/31 16:41:50 INFO scheduler.OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
18/01/31 16:41:50 INFO spark.SparkContext: Successfully stopped SparkContext
18/01/31 16:41:50 INFO util.ShutdownHookManager: Shutdown hook called
18/01/31 16:41:50 INFO util.ShutdownHookManager: Deleting directory /tmp/spark-4cf9bea8-662f-4c9b-97fc-f2ce8ba7a55b
Exception in thread "main" java.lang.Exception: spark-submit failed to complete properly given these arguments:
	--class com.ibm.sparktc.sparkbench.cli.CLIKickoff --executor-memory 2g --master spark://sparkmaster-sparkmaster-hdfssparkstandalone.marathon-user.containerip.dcos.thisdcos.directory:7077 /opt/spark-bench/lib/spark-bench-2.1.1_0.2.2-RELEASE.jar {"spark-bench":{"spark-submit-config":[{"spark-args":{"executor-memory":"2g"},"suites-parallel":false,"workload-suites":[{"benchmark-output":"hdfs:///tmp/csv-vs-parquet/results-data-gen3.csv","descr":"Generate a dataset, then take that same dataset and write it out to Parquet format","parallel":false,"repeat":10,"workloads":[{"cols":24,"name":"data-generation-kmeans","output":"hdfs:///tmp/csv-vs-parquet/kmeans-data3.csv","rows":10000000},{"input":"hdfs:///tmp/csv-vs-parquet/kmeans-data3.csv","name":"sql","output":"hdfs:///tmp/csv-vs-parquet/kmeans-data3.parquet","query":"select * from input"}]}]}]}}
	at com.ibm.sparktc.sparkbench.sparklaunch.SparkLaunch$.launch(SparkLaunch.scala:65)
	at com.ibm.sparktc.sparkbench.sparklaunch.SparkLaunch$$anonfun$launchSparkSubmitScripts$2.apply(SparkLaunch.scala:57)
	at com.ibm.sparktc.sparkbench.sparklaunch.SparkLaunch$$anonfun$launchSparkSubmitScripts$2.apply(SparkLaunch.scala:57)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at com.ibm.sparktc.sparkbench.sparklaunch.SparkLaunch$.launchSparkSubmitScripts(SparkLaunch.scala:57)
	at com.ibm.sparktc.sparkbench.sparklaunch.SparkLaunch$.main(SparkLaunch.scala:34)
	at com.ibm.sparktc.sparkbench.sparklaunch.SparkLaunch.main(SparkLaunch.scala)```
<!-- If running Spark-Bench threw a stack trace, paste it here. Please also surround this part in three backticks. -->

### Description of Problem, Any Other Info

If you use the repeat option in the configuration file and you have to specify an input or output file for the workload, the next iteration will fail because the file already exists.

<!-- Description and any other info here -->

**Comment by ecurtin** [comment]
Hi @Brandonage, thanks for your bug report!

I've made a couple of changes to address this issue. The biggest one is that I added a "save-mode" option for the config file where users can specify "errorifexists", "ignore", "overwrite", and for the benchmark output "append". How to do sensible appending for workloads, where each one is generating a new dataset, is an open question.

However, for your case, you'll be able to specify `save-mode = "overwrite"` in your workload and have each repetition of the workload write overwrite the results of the previous.
